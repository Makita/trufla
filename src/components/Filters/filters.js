/* eslint-disable no-invalid-this */
import React, { Component } from 'react';

import {
  ControlLabel,
  FormControl,
  Grid,
  Row,
  Col
} from 'react-bootstrap';

import style from './filters.scss';

const SelectOptions = ({ data }) => {
  return Array.from(data).map((item) => {
    // eslint-disable-next-line array-element-newline
    return <option key={item.id} value={item.id}>{item.name || item.code}</option>;
  })
};

interface FiltersProps {
  handleDepartmentFilterChange: (Event) => void
}

interface FiltersState {
  departments: Object,
  promoCodes: Object
}

export default class Filters extends Component<FiltersProps, FiltersState> {
  state = {
    departments: {},
    promoCodes: {}
  }

  componentDidMount() {
    fetch('/departments')
      .then(res => res.json())
      .then(departments => this.setState({ departments }));

    fetch('/promo_codes')
      .then(res => res.json())
      .then(promoCodes => this.setState({ promoCodes }));
  }

  render() {
    const { departments, promoCodes } = this.state;
    const {
      handleDepartmentFilterChange,
      handlePromoCodeChange,
      handleSearchChange
    } = this.props;

    return (
      <div className={style.container}>
        <Grid>
          <Row>
            <Col md={2}>
              <ControlLabel>Department</ControlLabel>
            </Col>
            <Col md={2}>
              <FormControl componentClass="select" placeholder="" onChange={handleDepartmentFilterChange}>
                <option value="">Unfiltered</option>
                <SelectOptions data={departments} />
              </FormControl>
            </Col>
            <Col md={2}>
              <ControlLabel>Promo Code</ControlLabel>
            </Col>
            <Col md={2}>
              <FormControl componentClass="select" placeholder="" onChange={handlePromoCodeChange}>
                <option value="">Unfiltered</option>
                <SelectOptions data={promoCodes} />
              </FormControl>
            </Col>
            <Col md={2}>
              <ControlLabel>Item Search</ControlLabel>
            </Col>
            <Col md={2}>
              <FormControl
                type="text"
                placeholder="Enter an item name to search for."
                onChange={handleSearchChange}
              />
            </Col>
          </Row>
        </Grid>
      </div>
    );
  }
}