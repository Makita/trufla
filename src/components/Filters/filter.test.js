/* eslint-disable no-undef */
import * as React from 'react';
import Filters, { SelectOptions } from './filters';

describe("SelectOptions", () => {
  it("creates three elements given an object with 3 properties", () => {
    const data = [
      { id: 1, name: "One" },
      { id: 2, name: "Two" },
      { id: 3, name: "Three" }
    ];

    const wrapper = mount(<SelectOptions data={data} />);

    expect(wrapper.children().length).toBe(data.length);
  });
})

describe("Filters", () => {
  it("should render correctly", () => {
    fetch
      .once(JSON.stringify([{ id: 1, name: "Department 1" }]))
      .once(JSON.stringify([{ id: 2, code: "Promo Code 1", discount: 100.0 }]));

    const wrapper = shallow(<Filters />)

    expect(wrapper).toMatchSnapshot();
  })
})