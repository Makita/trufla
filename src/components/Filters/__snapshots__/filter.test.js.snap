// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters should render correctly 1`] = `
<div
  className="container"
>
  <Grid
    bsClass="container"
    componentClass="div"
    fluid={false}
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        componentClass="div"
        md={2}
      >
        <ControlLabel
          bsClass="control-label"
          srOnly={false}
        >
          Department
        </ControlLabel>
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        md={2}
      >
        <FormControl
          bsClass="form-control"
          componentClass="select"
          placeholder=""
        >
          <option
            value=""
          >
            Unfiltered
          </option>
          <SelectOptions
            data={Object {}}
          />
        </FormControl>
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        md={2}
      >
        <ControlLabel
          bsClass="control-label"
          srOnly={false}
        >
          Promo Code
        </ControlLabel>
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        md={2}
      >
        <FormControl
          bsClass="form-control"
          componentClass="select"
          placeholder=""
        >
          <option
            value=""
          >
            Unfiltered
          </option>
          <SelectOptions
            data={Object {}}
          />
        </FormControl>
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        md={2}
      >
        <ControlLabel
          bsClass="control-label"
          srOnly={false}
        >
          Item Search
        </ControlLabel>
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        md={2}
      >
        <FormControl
          bsClass="form-control"
          componentClass="input"
          placeholder="Enter an item name to search for."
          type="text"
        />
      </Col>
    </Row>
  </Grid>
</div>
`;
